### DISCUSSION TITLE
DecentralizedStablecoin contract does not compile with the below error.

### DISCUSSION THREAD

**Discussion by helloram007** [Cyfrin/foundry-full-course-cu#2223] [Q&A]
Hi,

i have copied the code from cyfrin course repo and still the code does not compile with the below error.

[⠒] Compiling...
[⠆] Compiling 35 files with Solc 0.8.26
[⠰] Solc 0.8.26 finished in 2.11s
Error: 
Compiler run failed:
Error (3415): No arguments passed to the base constructor. Specify the arguments or mark "DecentralizedStableCoin" as abstract.
  --> src/DecentralizedStableCoin.sol:42:1:
   |
42 | contract DecentralizedStableCoin is ERC20Burnable, Ownable {
   | ^ (Relevant source part starts here and spans across multiple lines).
Note: Base constructor parameters:
  --> lib/openzeppelin-contracts/contracts/access/Ownable.sol:38:16:
   |
38 |     constructor(address initialOwner) {
   |                ^^^^^^^^^^^^^^^^^^^^^^

rusty@letsgetrusty-vm:~/cyfrin/defi/defi-stablecoin$

**Comment by Alessandro-Cavaliere** [comment]
Can you share `DecentralizedStableCoin` contract?

**Comment by urosognjenovic** [comment]
You need to pass the constructor argument (address initialOwner) to the Ownable contract.

For example: 

```
contract DecentralizedStableCoin is ERC20Burnable, Ownable(address(1))
```

**Reply by Lukelr12** [reply]
ran into the same problem, this solution worked perfectly.

**Reply by EngrPips** [reply]
Interesting.

**Comment by EngrPips** [comment]
Hello @helloram007, Please do as suggested by @thebraveman17  and let us know if the issue gets resolved.

**Comment by helloram007** [comment]
I had that i have pass something to Ownable, but Patricks code does not
have that, and when i inherited Ownable it didnot thrw any error, so left
it.
i will not pass the param and see if it works.
Thanks

**Reply by EngrPips** [reply]
It is not compulsory your code is exactly like Patrick's code. Most likely, in some parts of the codebase, your code will have to be different from his because the tools we use get updated from time to time, and some stuff might have changed, which you now need to account for to make your code work appropriately.

**Comment by AwnishSingh** [comment]
To follow along with the course, you need to install version 4.8.3 which can be done by

`forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit`
 
instead of
 
`forge install openzeppelin/openzeppelin-contracts --no-commit`

And then in the DecentralizedStableCoin.sol write the constructor like this

`constructor() ERC20("DecentralizedStableCoin", "DSC") {}`

**Reply by CHupinex** [reply]
Yes , its work.

**Reply by maktriss** [reply]
this solved my problem

**Reply by EngrPips** [reply]
Awesome.

**Reply by ygorz** [reply]
This helped my issue too, thank you!

**Comment by EngrPips** [comment]
Good job.
